libvast/src/format/json.cpp

```diff
@@ -268,6 +268,26 @@ caf::expected<data> type_biased_convert_impl<::simdjson::dom::object, map>(
   return xs;
 }
 
+template <>
+caf::expected<data> type_biased_convert_impl<::simdjson::dom::object, record>(
+  ::simdjson::dom::object o, const type& t) {
+  const auto& r = dynamic_cast<const record_type&>(*t);
+  record xs;
+  xs.reserve(o.size());
+  for (auto [k, v] : o) {
+    if (const auto* rf = r.find(k); !rf) {
```

Comment:
```suggestion
  ::simdjson::dom::object o, const type& t) {
  const auto* r = caf::get_if<const record_type>(&t);
  VAST_ASSERT(r != nullptr);
  record xs;
  xs.reserve(o.size());
  for (auto [k, v] : o) {
    if (const auto* rf = r->find(k); !rf) {
```

Any reason not to do this?

Reply:
No need to go through sum type access again, we already know at this point that we have a map. Also, if we change anything here, we should change it for all visitors at the same time.